### PR TITLE
unix,win: make on_alloc_cb failures more resilient

### PR DESCRIPTION
**Pull request by saghul**
Someone on IRC was confused about how to handle OOM failures on `alloc_cb` and called `uv_close` there. I think handling the NULL case makes the overall handling more resilient so I here we are. Also, initialized the `uv_buf_t` we pass around, for good measure.

R= @libuv/collaborators 
CI:  https://ci.nodejs.org/view/libuv/job/libuv-test-commit/93/

### CODE REVIEW

File: test/test-tcp-alloc-cb-fail.c [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  ASSERT(status == 0);
+}
+
+static void conn_alloc_cb(uv_handle_t* handle, size_t size, uv_buf_t* buf) {
+  /* Do nothing, read_cb should be called with UV_ENOBUFS. */
+}
+
+static void conn_read_cb(uv_stream_t* stream,
+                           ssize_t nread,
+                           const uv_buf_t* buf) {
```
Comment (bnoordhuis) [inline test/test-tcp-alloc-cb-fail.c]: Indentation is off here.

**Comment by bnoordhuis** [comment]
LGTM with a style nit.  I confess to only having glanced at the tests but I think you copied them from existing tests anyway so it's probably alright. :-)

**Comment by saghul** [comment]
Landed. Yeah, the tests are adapted from other tests :-)

**Comment by indutny** [comment]
Belated LGTM

**Pull request merged**